lib/article_json/export/amp/elements/base.rb

```diff
@@ -0,0 +1,52 @@
+module ArticleJSON
+  module Export
+    module AMP
+      module Elements
+        class Base
+          # @param [ArticleJSON::Elements::Base] element
+          def initialize(element)
+            @element = element
+          end
+
+          # Export a HTML node out of the given element
+          # Dynamically looks up the right export-element-class, instantiates it
+          # and then calls the #build method.
+          # @return [Nokogiri::HTML::Node]
+          def export
+            exporter = self.class == Base ? self.class.build(@element) : self
+            exporter.export unless exporter.nil?
+          end
+
+          private
+
+          def create_element(tag, *args)
+            Nokogiri::HTML.fragment('').document.create_element(tag.to_s, *args)
```

Comment:
I guess we can share this with the HTML exporter via module? same goes for `#create_text_node`...

Reply:
yeah, that makes sense...

Reply:
trying that, I'm closing this PR, since it's a different approach.